frontend/test/model/common/SchemaButton.spec.tsx

```diff
@@ -0,0 +1,35 @@
+import { ThemeProvider } from '@mui/material/styles'
+import { render, screen, waitFor } from '@testing-library/react'
+import { MarkdownDisplayProps } from 'src/common/MarkdownDisplay'
+import SchemaButton from 'src/model/beta/common/SchemaButton'
+import { lightTheme } from 'src/theme'
+import { testAccessRequestSchema } from 'utils/test/testModels'
+import { describe, expect, it, vi } from 'vitest'
+
+vi.mock('src/common/MarkdownDisplay.tsx', () => ({ default: (_props: MarkdownDisplayProps) => <></> }))
+
+describe('SchemaButton', () => {
+  it('renders a SchemaButton component', async () => {
+    render(
+      <ThemeProvider theme={lightTheme}>
+        <SchemaButton schema={testAccessRequestSchema} onClick={() => undefined} />
+      </ThemeProvider>,
+    )
+
+    await waitFor(async () => {
+      expect(await screen.findByText('Select schema')).not.toBeUndefined()
+    })
+  })
+
+  it('renders a loading SchemaButton component', async () => {
```

Comment:
This is a decent unit test, but it would be good to follow a naming convention such as "Something is true/false when something is the case". So here it would be `it('enters a loading state when the loading prop is true').